### DISCUSSION TITLE
Not able to connect multiple ue

### DISCUSSION THREAD

**Discussion by aritra-nandy** [HewlettPackard/PacketRusher#83] [Q&A] [answered]
I have configured multiple UEs in the configuration file but when hits multi-ue -n 2 command it's not working for me showing "Interface for UE 0000000022 has not been created. Tunnel has been disabled" . Can anyone help me how to connect multiple UEs and what should be the correct configuration?

I have used the below configuration file 
[packetRusher_config.txt](https://github.com/HewlettPackard/PacketRusher/files/14382539/packetRusher_config.txt)

**Comment by linouxis9** [comment]
Hi @aritra-nandy!
You can follow here for more information regarding the usage of multi-ue: https://github.com/HewlettPackard/PacketRusher/issues/65#issuecomment-1895554491

**Comment by aritra-nandy** [comment]
Can you explain how to use --tunnel and --dedicatedGnb option briefly?

**Comment by linouxis9** [comment, marked as answer]
You need to run `sudo ./packetrusher multi-ue -n 2 --tunnel --dedicatedGnb` @aritra-nandy
For --tunnel to work with multiple UEs, you must have a dedicated gNodeB per UE, else --tunnel does not work. --dedicatedGnb does that.

PacketRusher automatically creates a second gNodeB with the same configuration as the first gNodeB for the second UE when using --dedicatedGnb, but with the first IP next to the first gNodeB: if the first gNodeB has IP 10.228.118.180, the second gNodeB created automatically by PacketRusher will have IP 10.228.118.181 and this cannot be configured differently for now.
As such, make sure to add the IP 10.228.118.181 to your network interface (eg. sudo ip addr add eth0 10.228.118.181)

**Comment by aritra-nandy** [comment]
Thank you for your help, able to connect two UE now

**Reply by linouxis9** [reply]
Great! Glad I could help. 
Cheers,
Valentin

**Comment by onikbakht** [comment]
Hello Valentin @linouxis9 
for using both of these different options below

`multi-ue -n 2 --tunnel-vrf -t --dedicatedGnb`
`multi-ue -n 2 --tunnel --dedicatedGnb`
I get the following error logs, which I could not figure out, what is causing it, and how to solve it.

`
time="2025-02-21T23:20:24Z" level=info msg="Selecting 10.33.33.9 for host amf.5gc.mnc002.mcc262.3gppnetwork.org as AMF's IP address"
time="2025-02-21T23:20:24Z" level=info msg="Selecting 10.33.33.20 for host gnb.packetrusher.org as gNodeB's N3/Data IP address"
time="2025-02-21T23:20:24Z" level=info msg="Selecting 10.33.33.20 for host gnb.packetrusher.org as gNodeB's N2/Control IP address"
time="2025-02-21T23:20:24Z" level=info msg="Loaded config at: /PacketRusher/config/packetrusher.yaml"
time="2025-02-21T23:20:24Z" level=info msg="PacketRusher version 1.0.1"
time="2025-02-21T23:20:24Z" level=info msg=---------------------------------------
time="2025-02-21T23:20:24Z" level=info msg="[TESTER] Starting test function: Testing registration of multiple UEs"
time="2025-02-21T23:20:24Z" level=info msg="[TESTER][UE] Number of UEs: 2"
time="2025-02-21T23:20:24Z" level=info msg="[TESTER][GNB] gNodeB control interface IP/Port: 10.33.33.20/38412~"
time="2025-02-21T23:20:24Z" level=info msg="[TESTER][GNB] gNodeB data interface IP/Port: 10.33.33.20/2152"
time="2025-02-21T23:20:24Z" level=info msg="[TESTER][AMF] AMF IP/Port: 10.33.33.9/38412"
time="2025-02-21T23:20:24Z" level=info msg=---------------------------------------
time="2025-02-21T23:20:24Z" level=info msg="[GNB] SCTP/NGAP service is running"
time="2025-02-21T23:20:24Z" level=info msg="[GNB] Initiating NG Setup Request"
time="2025-02-21T23:20:24Z" level=fatal msg="Error incannot assign requested address"
`

same error logs for both options of running the Packetrusher.

I look for ward to your kind reply

Best Regards
Omid
